app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/database/Database.java

```diff
@@ -1,9 +1,48 @@
 package com.epfl.neighborfood.neighborfoodandroid.database;
 
-import com.epfl.neighborfood.neighborfoodandroid.models.Model;
-
-import java.util.List;
+import com.google.android.gms.tasks.Task;
 
 public interface Database {
 
+
+    /**
+     * Fetches the requested data
+     * @param collectionPath collection
+     * @param documentPath document id
+     * @return task that fails if the database is unreachable
+     */
+    Task<DocumentSnapshot> fetch(String collectionPath, String documentPath);
```

Comment:
Using DocumentSnpashot as the return type of the Task might be an abstraction leak, as these types are specific to firebase. I dont think its worth it for now to switch to a more abstract implementation, though